clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -3100,7 +3100,11 @@ void handleResponse(AbstractResponse abstractResponse) {
                         final Long offset = partitionData.offset;
                         final String metadata = partitionData.metadata;
                         final Optional<Integer> leaderEpoch = partitionData.leaderEpoch;
-                        groupOffsetsListing.put(topicPartition, new OffsetAndMetadata(offset, leaderEpoch, metadata));
+                        if (offset < 0) {
```

Comment:
maybe add a comment here that negative offset indicates there is no committed offset for the topic partition? Would be easier for other people to understand the code.